Failed to parse ONNX model on FaceRecognizerSF

Error when trying to run face recognition in Livebook

<img width="763" alt="スクリーンショット 2023-07-05 22 43 43" src="https://github.com/cocoa-xu/evision/assets/14919320/02073214-4d01-4e2b-9f5a-6680a1ae3723">

```
13:42:07.562 [error] Kino.listen with #Function<44.65746770/1 in :erl_eval.expr/5> failed with reason:

** (FunctionClauseError) no function clause matching in Evision.FaceRecognizerSF.feature/2
    (evision 0.1.31) lib/generated/evision_facerecognizersf.ex:231: Evision.FaceRecognizerSF.feature({:error, "OpenCV(4.7.0) /home/livebook/.cache/mix/installs/elixir-1.14.2-erts-12.3.2.2/a6208935cb3fad81dd0c837b0fb54863/deps/evision/3rd_party/opencv/opencv-4.7.0/modules/dnn/src/onnx/onnx_importer.cpp:275: error: (-210:Unsupported format or combination of formats) Failed to parse ONNX model: /home/livebook/.cache/face_recognition_sface_2021dec.onnx in function 'ONNXImporter'\n"}, {:error, "cannot get `Ptr<cv::FaceRecognizerSF>` from `self`: mismatched type or invalid resource?"})
    (stdlib 3.17.2) erl_eval.erl:685: :erl_eval.do_apply/6
    (stdlib 3.17.2) erl_eval.erl:446: :erl_eval.expr/5
    (stdlib 3.17.2) erl_eval.erl:123: :erl_eval.exprs/5
    (kino 0.9.4) lib/kino.ex:407: Kino.safe_apply/3
    (elixir 1.14.2) lib/enum.ex:980: anonymous fn/3 in Enum.each/2
    (elixir 1.14.2) lib/enum.ex:4307: anonymous fn/3 in Enum.each/2
    (elixir 1.14.2) lib/stream.ex:1799: anonymous fn/3 in Enumerable.Stream.reduce/3
```

It appears that the model file has not been downloaded correctly.
The model file face_recognition_sface_2021dec.onnx appears to be HTML.

https://github.com/cocoa-xu/evision/blob/7369af18373bf885d86d9a50f9757ff63cad775b/lib/zoo/face_recognition/sface.ex#L325

Probably because the branch name of opencv_zoo was changed from master to main, it is not downloaded correctly.

--- a/lib/zoo/face_recognition/sface.ex
+++ b/lib/zoo/face_recognition/sface.ex
@@ -322,14 +322,14 @@ defmodule Evision.Zoo.FaceRecognition.SFace do
   @spec model_info(:default_model | :quant_model) :: {String.t(), String.t()}
   def model_info(:default_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/face_recognition_sface/face_recognition_sface_2021dec.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/face_recognition_sface/face_recognition_sface_2021dec.onnx?raw=true",
       "face_recognition_sface_2021dec.onnx"
     }
   end
 
   def model_info(:quant_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/face_recognition_sface/face_recognition_sface_2021dec-act_int8-wt_int8-quantized.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/face_recognition_sface/face_recognition_sface_2021dec-act_int8-wt_int8-quantized.onnx?raw=true",
       "face_recognition_sface_2021dec-act_int8-wt_int8-quantized.onnx"
     }
   end

--- a/lib/zoo/image_classification/mobilenet_v1.ex
+++ b/lib/zoo/image_classification/mobilenet_v1.ex
@@ -208,14 +208,14 @@ defmodule Evision.Zoo.ImageClassification.MobileNetV1 do
   @spec model_info(:default_model | :quant_model) :: {String.t(), String.t()}
   def model_info(:default_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/image_classification_mobilenet/image_classification_mobilenetv1_2022apr.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/image_classification_mobilenet/image_classification_mobilenetv1_2022apr.onnx?raw=true",
       "image_classification_mobilenetv1_2022apr.onnx"
     }
   end
 
   def model_info(:quant_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/image_classification_mobilenet/image_classification_mobilenetv1_2022apr-int8-quantized.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/image_classification_mobilenet/image_classification_mobilenetv1_2022apr-int8-quantized.onnx?raw=true",
       "image_classification_mobilenetv1_2022apr-int8-quantized.onnx"
     }
   end

--- a/lib/zoo/image_classification/pp_resnet.ex
+++ b/lib/zoo/image_classification/pp_resnet.ex
@@ -209,14 +209,14 @@ defmodule Evision.Zoo.ImageClassification.PPResNet do
   @spec model_info(:default_model | :quant_model) :: {String.t(), String.t()}
   def model_info(:default_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/image_classification_ppresnet/image_classification_ppresnet50_2022jan.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/image_classification_ppresnet/image_classification_ppresnet50_2022jan.onnx?raw=true",
       "image_classification_ppresnet50_2022jan.onnx"
     }
   end
 
   def model_info(:quant_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/image_classification_ppresnet/image_classification_ppresnet50_2022jan-act_int8-wt_int8-quantized.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/image_classification_ppresnet/image_classification_ppresnet50_2022jan-act_int8-wt_int8-quantized.onnx?raw=true",
       "image_classification_ppresnet50_2022jan-act_int8-wt_int8-quantized.onnx"
     }
   end

--- a/lib/zoo/image_segmentation/pp_humanseg.ex
+++ b/lib/zoo/image_segmentation/pp_humanseg.ex
@@ -183,14 +183,14 @@ defmodule Evision.Zoo.ImageSegmentation.PPHumanSeg do
   @spec model_info(:default_model | :quant_model) :: {String.t(), String.t()}
   def model_info(:default_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/human_segmentation_pphumanseg/human_segmentation_pphumanseg_2021oct.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/human_segmentation_pphumanseg/human_segmentation_pphumanseg_2021oct.onnx?raw=true",
       "human_segmentation_pphumanseg_2021oct.onnx"
     }
   end
 
   def model_info(:quant_model) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/human_segmentation_pphumanseg/human_segmentation_pphumanseg_2021oct-act_int8-wt_int8-quantized.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/human_segmentation_pphumanseg/human_segmentation_pphumanseg_2021oct-act_int8-wt_int8-quantized.onnx?raw=true",
       "human_segmentation_pphumanseg_2021oct-act_int8-wt_int8-quantized.onnx"
     }
   end

--- a/lib/zoo/text_detection/db.ex
+++ b/lib/zoo/text_detection/db.ex
@@ -187,7 +187,7 @@ defmodule Evision.Zoo.TextDetection.DB do
   @spec model_info(:default_model | :quant_model) :: {String.t(), String.t()}
   def model_info(:ic15_resnet18) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/text_detection_db/text_detection_DB_IC15_resnet18_2021sep.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/text_detection_db/text_detection_DB_IC15_resnet18_2021sep.onnx?raw=true",
       "text_detection_DB_IC15_resnet18_2021sep.onnx"
     }
   end
@@ -202,7 +202,7 @@ defmodule Evision.Zoo.TextDetection.DB do
 
   def model_info(:td500_resnet18) do
     {
-      "https://github.com/opencv/opencv_zoo/blob/master/models/text_detection_db/text_detection_DB_TD500_resnet18_2021sep.onnx?raw=true",
+      "https://github.com/opencv/opencv_zoo/blob/main/models/text_detection_db/text_detection_DB_TD500_resnet18_2021sep.onnx?raw=true",
       "text_detection_DB_TD500_resnet18_2021sep.onnx"
     }
   end

